pkg/llm-d-inference-sim/streaming.go

```diff
@@ -60,7 +60,11 @@ func (s *VllmSimulator) sendStreamingResponse(context *streamingContext, respons
 					s.sendTokenChunks(context, w, tc.Function.tokenizedArguments, &tc, finishReason)
 				}
 			} else {
-				s.logger.Info("Going to send text", "number of tokens", usageData.CompletionTokens)
+				if usageData != nil {
+					s.logger.Info("Going to send text", "number of tokens", usageData.CompletionTokens)
+				} else {
+					s.logger.Info("Going to send text", "number of tokens", len(responseTokens))
```

Comment:
I think we can simply always print len(responseTokens) here without checking if usageData exists

Reply:
that makes sense

Reply:
Thanks for fixing the crash!